pkg/daemon/update.go

```diff
@@ -2120,6 +2120,28 @@ func (dn *CoreOSDaemon) applyLayeredOSChanges(mcDiff machineConfigDiff, oldConfi
 		defer os.Remove(extensionsRepo)
 	}
 
+	// Always clean up pending, because the RT kernel switch logic below operates on booted,
+	// not pending.
+	if err := removePendingDeployment(); err != nil {
```

Comment:
In what ways can `cleanup -p` fail? I assume it won't error if there isn't a pending deployment?

Reply:
yeah, it doesn't give error. Tested it on local machine
```
$ rpm-ostree cleanup -p
Deployments unchanged.
$ echo $?
0
```

Reply:
Right, it's idempotent.  Also crucially, this code path is executed by CI, so if it didn't work, CI would fail.